op-node/rollup/engine/events.go

```diff
@@ -40,6 +40,54 @@ func (ev ForkchoiceUpdateEvent) String() string {
 	return "forkchoice-update"
 }
 
+// PromoteUnsafeEvent signals that the given block may now become a canonical unsafe block.
+// This is pre-forkchoice update; the change may not be reflected yet in the EL.
+// Warning: this even does not always fire;
```

Comment:
```suggestion
// Warning: this event does not always fire;
```

I could also use more explanation about under which conditions it does not fire, the comment below about "the legacy pre-events unsafe block processing does not emit this" doesn't explain when it does/doesn't emit.

Reply:
It was a stale comment, the duplicate legacy code-path does emit the event now. I still left a note about the duplication, something I'd like to clean up eventually.